### DISCUSSION TITLE
weblate does not automatically get the files that I push in the git repository

### DISCUSSION THREAD

**Discussion by mouradbenfradj** [WeblateOrg/weblate#7001] [Q&A] [answered]
When I create a weblate addon (component discovery) with the REST api,

**'curl -X POST -H "Content-Type: application/json" ' +
'-H "Authorization: Token **************" ' +
'-k -d "{' +
'\\"name\\":\\"weblate.discovery.discovery\\",' +
'\\"configuration\\":{' +
'\\"match\\":\\"locales/(?P^<language^>[^^/]*)/(?P^<component^>(?!__TEMPLATE__) ([^^/])*)\\\\.json\\",' +
'\\"file_format\\":\\"i18next\\",' +
'\\"name_template\\":\\"{{ component }}_fr\\",' +
'\\"base_file_template\\":\\"locales/fr/{{component}}.json\\",' +
'\\"new_base_template\\":\\"locales/fr/{{component}}.json\\",' +
'\\"language_regex\\":\\"^^[^^.]+$\\",' +
'\\"copy_addons\\":true,' +
'\\"remove\\":false,' +
'\\"confirm\\":true,' +
'\\"preview\\":true},' +
'\\"url\\":\\"\\"' +
'}" http://127.0.0.1/api/components/repositoryname/__template__fr/addons/'**
, weblate does not automatically get the files that I push in the git repository.

To make it work, I first need to manually 

-1) go to the discovery addon webpage in the web browser, 
-2) save the addon in the web browser 
-3) accept the list of files that match the template in the web browser(see attached image). 
![manualtiget](https://user-images.githubusercontent.com/17617401/147467884-5bdecbe6-5bea-438b-a5f9-94c635909020.png)

Once this has been done one time, all the files that I push in the git repository are correctly discovered by weblate. However we would like to do the 1), 2) & 3) steps automaticaly with the REST API. I tried to set the confirm and preview fields of our CURL request to false but it does not work. Is there a way to configure the CURL request so that weblate discover the files without any manual steps?

**Weblate versions**

Weblate 4.5.3 
Django 3.1.8 
siphashc 2.1 
translate-toolkit 3.3.4 
lxml 4.6.3 
Pillow 8.1.2 
bleach 3.3.0 
python-dateutil 2.8.1 
social-auth-core 4.1.0 
social-auth-app-django 4.0.0 
django-crispy-forms 1.11.2 
oauthlib 3.1.0 
django-compressor 2.4 
djangorestframework 3.12.4 
django-filter 2.4.0 
django-appconf 1.0.4 
user-agents 2.2.0 
filelock 3.0.12 
setuptools 40.8.0 
jellyfish 0.8.2 
openpyxl 3.0.7 
celery 5.0.5 
kombu 5.0.2 
translation-finder 2.9 
weblate-language-data 2021.3 
html2text 2020.1.16 
pycairo 1.16.2 
pygobject 3.30.4 
diff-match-patch 20200713 
requests 2.25.1 
django-redis 4.12.1 
hiredis 2.0.0 
sentry_sdk 1.0.0 
Cython 0.29.22 
misaka 2.1.1 
GitPython 3.1.14 
borgbackup 1.1.16 
pyparsing 2.4.7 
pyahocorasick 1.4.2 
Python 3.7.3 
Git 2.20.1 
psycopg2 2.8.6 
psycopg2-binary 2.8.6 
phply 1.2.5 
chardet 4.0.0 
ruamel.yaml 0.17.4 
tesserocr 2.5.1 
akismet 1.1 
boto3 1.17.50 
zeep 4.0.0 
aeidon 1.9 
iniparse 0.5 
mysqlclient 2.0.3 
Mercurial 5.7.1 
git-svn 2.20.1 
git-review 2.0.0 
Redis server 6.2.1 
PostgreSQL server 13.2

**Comment by nijel** [comment, marked as answer]
Copying and pasting answer for copied and pasted question at https://stackoverflow.com/a/70507178/225718:

It should work. I've [just added](https://github.com/WeblateOrg/weblate/commit/da1ba0986becab697fa034ef94115b598473f45a) test to cover this to our test suite.

My guess would be that something is wrong in the JSON request - maybe something is not properly escaped or is escaped too much?

What does the curl request end with? ~~What Weblate version do you use?~~

PS: You need to upgrade, this has been fixed in b9646f23174

**Reply by mouradbenfradj** [reply]
Thanks for your answer .
I have upgrade weblate to the last version and now it work.
Thanks .
